packages/components/src/box-model-overlay/index.tsx

```diff
@@ -0,0 +1,271 @@
+/**
+ * External dependencies
+ */
+import styled from '@emotion/styled';
+import type { ReactElement, RefObject } from 'react';
+
+/**
+ * WordPress dependencies
+ */
+import {
+	useRef,
+	useEffect,
+	useMemo,
+	useCallback,
+	forwardRef,
+	useImperativeHandle,
+	cloneElement,
+	Children,
+} from '@wordpress/element';
+
+/**
+ * Internal dependencies
+ */
+import Popover from '../popover';
+
+type BoxModelSides = 'top' | 'right' | 'bottom' | 'left';
+
+interface BoxModelOverlayHandle {
+	update: () => void;
+}
+
+interface BoxModelOverlayBaseProps {
+	showValues: {
+		margin?: {
+			[ side in BoxModelSides ]?: boolean;
+		};
+		padding?: {
+			[ side in BoxModelSides ]?: boolean;
+		};
+	};
+}
+interface BoxModelOverlayPropsWithTargetRef extends BoxModelOverlayBaseProps {
+	targetRef: RefObject< HTMLElement >;
+}
+interface BoxModelOverlayPropsWithChildren extends BoxModelOverlayBaseProps {
+	children: ReactElement;
+}
+type BoxModelOverlayProps =
+	| BoxModelOverlayPropsWithTargetRef
+	| BoxModelOverlayPropsWithChildren;
+
+const DEFAULT_SHOW_VALUES: BoxModelOverlayBaseProps[ 'showValues' ] = {};
+
+// Copied from Chrome's DevTools: https://github.com/ChromeDevTools/devtools-frontend/blob/088a8f175bd58f2e0e2d492e991a3253124d7c11/front_end/core/common/Color.ts#L931
+const MARGIN_COLOR = 'rgba( 246, 178, 107, 0.66 )';
+// Copied from Chrome's DevTools: https://github.com/ChromeDevTools/devtools-frontend/blob/088a8f175bd58f2e0e2d492e991a3253124d7c11/front_end/core/common/Color.ts#L927
+const PADDING_COLOR = 'rgba( 147, 196, 125, 0.55 )';
```

Comment:
These values mimic the styles in Chrome's DevTools. Not sure which WordPress-related colors we can use here? Ideas welcome!

Reply:
If we are looking for specific WordPress colors, perhaps the [Color docs from the style guide](https://make.wordpress.org/design/handbook/design-guide/foundations/colors/) might help. There's also a [codepen with full list of official colors](https://codepen.io/hugobaeta/full/RNOzoV/).